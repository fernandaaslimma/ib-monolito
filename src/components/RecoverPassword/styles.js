import styled, { css } from "styled-components";
import { rem, remFontSize, media } from "../../styles/tools";
import {
  grey50,
  grey10,
  black50,
  red,
  lightestBlue20
} from "../../styles/settings";

export const BackButtonWrapper = styled.div`
  display: inline-flex;
  opacity: 1;
  cursor: pointer;
  transform: translateY(${rem(-15)});
  ${media.md(css`
    color: ${lightestBlue20};
    letter-spacing: ${rem(0.4)};
    font-size: ${remFontSize(11)};
    font-weight: 600;
  `)};
`;

export const Title = styled.h2`
  font-family: "Roboto Bold", "Roboto Medium", Roboto;
  font-size: ${remFontSize(20)};
  letter-spacing: 0.1px;
  color: ${grey10};

  ${media.md(css`
    margin-bottom: ${rem(10)};
  `)};
`;

export const Subtitle = styled.h3`
  font-family: Lato;
  margin-bottom: ${rem(20)};
  font-size: ${remFontSize(13)};
  font-style: italic;
  line-height: 1.23;
  letter-spacing: ${rem(0.3)};
  text-align: center;
  color: ${black50};
  margin-top: ${rem(8)};
  ${media.xs(css`
    margin-top: ${rem(1)};
  `)};
`;

export const ButtonWrapper = styled.div`
  margin: ${rem(23)} 0 ${rem(32)};
`;

export const IconWrapper = styled.div`
  padding: ${rem(27)} ${rem(34)} ${rem(3)} ${rem(44)};

  ${media.md(css`
    padding-bottom: 0;
  `)};
`;

export const Header = styled.div`
  text-align: center;
  border-bottom: solid ${rem(0.7)} ${grey50};
  min-height: ${rem(83)};
`;

export const HeaderName = styled.p`
  font-family: "Lato";
  font-size: ${remFontSize(14)};
  line-height: 1;
  letter-spacing: ${rem(0.3)};
  text-align: center;
  color: ${black50};
  padding-top: ${rem(6)};
  word-break: break-all;

  ${media.md(css`
    line-height: 2.42;
  `)};
`;

export const HeaderEmail = styled(HeaderName)`
  padding-bottom: ${rem(6)};

  ${media.md(css`
    margin-top: -${rem(18)};
  `)};
`;

export const ErrorMessage = styled.p`
  font-family: Lato;
  font-size: ${remFontSize(13)};
  font-style: italic;
  letter-spacing: ${rem(0.1)};
  text-align: center;
  color: ${red};
  height: ${rem(16)};
  margin-bottom: ${rem(9)};

  ${media.md(css`
    margin-bottom: ${rem(28)};
  `)};
`;
