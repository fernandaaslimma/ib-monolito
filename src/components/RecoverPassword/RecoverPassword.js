import React, { Fragment, Component } from "react";
import { Redirect } from "react-router-dom";
import Icon from "../common/Icon/";
import Link from "../common/Link";
import { string, func, shape, bool } from "prop-types";
import { translate } from "../../utils/i18n";
import { checkValidEmail } from "../../utils/validations/login";
import { isInternetExplorer } from "../../utils/getNavigator";
import { preflightForInternetExplorer } from "../../services/login";

import Input from "../common/Input";
import Button from "../common/Button";
import {
  Header,
  IconWrapper,
  Title,
  ButtonWrapper,
  BackButtonWrapper,
  Subtitle
} from "./styles";

import { Wrapper, Content, Container } from "../Login/styles";

class RecoverPassword extends Component {
  constructor(props) {
    super(props);

    this.isSubmitDisabled = this.isSubmitDisabled.bind(this);
    this.isValidEmail = this.isValidEmail.bind(this);
  }
  componentWillMount() {
    /*
    This request is required for Internet Explorer only
    It makes a preflight request without the headers.
    */
    if (isInternetExplorer()) {
      preflightForInternetExplorer();
    }
  }

  isSubmitDisabled(isValidEmail, email) {
    return isValidEmail && email.length < 4;
  }

  isValidEmail(email) {
    return checkValidEmail(email) || null;
  }

  render() {
    const {
      handleUserInputRecoverPassword,
      recoverPassword,
      handleUserSubmit,
      cleanPasswordSuccess,
      recoverPasswordSuccess
    } = this.props;
    if (recoverPasswordSuccess) {
      return (
        <Redirect
          to={{
            pathname: "/email-sent",
            search: `?email=${recoverPassword.email}`
          }}
        />
      );
    }
    return (
      <Fragment>
        <Container>
          <Wrapper data-test="RecoverPassword">
            <Header>
              <IconWrapper>
                <Icon type="BrandLogo" />
              </IconWrapper>
            </Header>
            <Content>
              <Title>{translate("RECOVER_PASSWORD")}</Title>
              <Subtitle>{translate("INSERT_EMAIL_TO_SEND_PASSWORD")}</Subtitle>
              <form onSubmit={handleUserSubmit} noValidate>
                <Input
                  icon="Email"
                  type="text"
                  name="email"
                  onChange={handleUserInputRecoverPassword}
                  autoComplete="new-password"
                  dataTest="RecoveryEmail"
                  label={translate("EMAIL")}
                  value={recoverPassword.email}
                  valid={() => this.isValidEmail(recoverPassword.email)}
                />
                <ButtonWrapper>
                  <Button
                    disabled={!this.isValidEmail(recoverPassword.email)}
                    opacity={0.5}
                    dataTest="SendButton"
                    isCallToAction
                    width="180"
                    type="submit"
                  >
                    {translate("SEND")}
                  </Button>
                </ButtonWrapper>
                <BackButtonWrapper>
                  <Link
                    to={"/"}
                    anchor
                    noSpan
                    opacity={1}
                    dataTest="BackToHomeScreen"
                    onClick={cleanPasswordSuccess}
                  >
                    {translate("BACK_TO_LOGIN")}
                  </Link>
                </BackButtonWrapper>
              </form>
            </Content>
          </Wrapper>
        </Container>
      </Fragment>
    );
  }
}

RecoverPassword.defaultProps = {
  recoverPassword: {
    email: ""
  },
  recoverPasswordSuccess: false
};

RecoverPassword.propTypes = {
  handleUserInputRecoverPassword: func.isRequired,
  handleUserSubmit: func.isRequired,
  recoverPassword: shape({
    email: string
  }),
  recoverPasswordSuccess: bool
};

export default RecoverPassword;
