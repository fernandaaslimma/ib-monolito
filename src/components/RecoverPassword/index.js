import { connect } from "redux-zero/react";

import RecoverPassword from "./RecoverPassword";
import actions from "../../actions/recoverPassword";

export default connect(
  ({ recoverPassword, recoverPasswordSuccess, cleanPasswordSuccess }) => ({
    recoverPassword,
    recoverPasswordSuccess,
    cleanPasswordSuccess
  }),
  actions
)(RecoverPassword);
