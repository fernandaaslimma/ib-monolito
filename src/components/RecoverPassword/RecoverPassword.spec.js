import React from "react";
import { shallow } from "enzyme";
import { Redirect } from "react-router-dom";
import RecoverPassword from "./RecoverPassword";
const handleUserInputRecoverPasswordMock = jest.fn();

const props = {
  recoverPassword: {
    error: null,
    email: ""
  },
  cleanPasswordSuccess: false,
  recoverPasswordSuccess: false,
  handleUserSubmit: () => {},
  handleUserInputRecoverPassword: () => {}
};

describe("RecoverPassword component", () => {
  it("should match snapshot with default props", () => {
    expect(shallow(<RecoverPassword {...props} />)).toMatchSnapshot();
  });
  it("should match snapshot with handleUserInput props", () => {
    expect(
      shallow(
        <RecoverPassword
          handleUserInputRecoverPassword={handleUserInputRecoverPasswordMock}
        />
      )
    ).toMatchSnapshot();
  });

  it("should match snapshot with email props", () => {
    expect(
      shallow(
        <RecoverPassword
          recoverPassword={{ email: "te000steee@asdas.com" }}
          handleUserInputRecoverPassword={handleUserInputRecoverPasswordMock}
        />
      )
    ).toMatchSnapshot();
  });
});

describe("isSubmitDisabled", () => {
  it("should return false when createPassword is empty", () => {
    const email = "";
    const instance = shallow(
      <RecoverPassword
        handleUserInputRecoverPassword={handleUserInputRecoverPasswordMock}
      />
    ).instance();
    expect(instance.isSubmitDisabled(false, email)).toBe(false);
  });
});

describe("isValidEmail", () => {
  it("should return False when email is not valid", () => {
    const instance = shallow(
      <RecoverPassword
        handleUserInputRecoverPassword={handleUserInputRecoverPasswordMock}
      />
    ).instance();
    expect(instance.isValidEmail("te000steee@")).toBe(null);
  });
  it("should return True when email is valid", () => {
    const instance = shallow(
      <RecoverPassword
        handleUserInputRecoverPassword={handleUserInputRecoverPasswordMock}
      />
    ).instance();
    expect(instance.isValidEmail("newBlo@email.com")).toBe(true);
  });
});

describe("recoverPassword is set to true", () => {
  it("should return to the correct pathname", () => {
    const newProps = {
      recoverPassword: {
        error: null,
        email: ""
      },
      cleanPasswordSuccess: false,
      recoverPasswordSuccess: {
        to: {
          pathname: "/email-sent",
          search: "?email="
        }
      },
      handleUserSubmit: () => {},
      handleUserInputRecoverPassword: () => {}
    };
    const wrapped = shallow(
      <RecoverPassword {...newProps} recoverPasswordSuccess={true} />
    );
    expect(wrapped.find(Redirect).props("recoverPasswordSuccess")).toEqual({
      push: false,
      to: {
        pathname: "/email-sent",
        search: "?email="
      }
    });
  });
});

// describe("isInternetExplorer", () => {
//   it("Should return true for msSaveOrOpenBlob when exist", () => {
//     dependency.isInternetExplorer = jest.fn(); //eslint-disable-line
//     // isInternetExplorer();
//     expect(shallow(<RecoverPassword recoverPasswordSuccess={true} />)).toBe(
//       true
//     );
//   });
// });
