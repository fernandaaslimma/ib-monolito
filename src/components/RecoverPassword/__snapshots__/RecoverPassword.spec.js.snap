// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecoverPassword component should match snapshot with default props 1`] = `
<Fragment>
  <styled.div>
    <styled.div
      data-test="RecoverPassword"
    >
      <styled.div>
        <styled.div>
          <Icon
            color={null}
            cursor={false}
            fixCrossAxisAlign={null}
            height={null}
            type="BrandLogo"
            width={null}
          />
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.h2 />
        <styled.h3 />
        <form
          noValidate={true}
          onSubmit={[Function]}
        >
          <Input
            EnableFixedInput={null}
            autoComplete="new-password"
            dataTest="RecoveryEmail"
            disableKeys={Array []}
            disabled={false}
            fixedValue={null}
            icon="Email"
            label={null}
            msgError={null}
            msgFocusInput={null}
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={null}
            placeholder=""
            tinyLabels={null}
            type="text"
            valid={[Function]}
            value=""
          />
          <styled.div>
            <Button
              actionSecondary={null}
              className={null}
              dataTest="SendButton"
              disabled={true}
              height={null}
              isCallToAction={true}
              isWarning={null}
              onClick={null}
              opacity={0.5}
              paddingSize={null}
              small={false}
              type="submit"
              width="180"
            />
          </styled.div>
          <styled.div>
            <Link
              actionSecondary={false}
              anchor={true}
              dataTest="BackToHomeScreen"
              disabled={false}
              href={null}
              isCallToAction={null}
              loading={false}
              noSpan={true}
              onClick={false}
              opacity={1}
              small={null}
              target={null}
              to="/"
              width={null}
            />
          </styled.div>
        </form>
      </styled.div>
    </styled.div>
  </styled.div>
</Fragment>
`;

exports[`RecoverPassword component should match snapshot with email props 1`] = `
<Fragment>
  <styled.div>
    <styled.div
      data-test="RecoverPassword"
    >
      <styled.div>
        <styled.div>
          <Icon
            color={null}
            cursor={false}
            fixCrossAxisAlign={null}
            height={null}
            type="BrandLogo"
            width={null}
          />
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.h2 />
        <styled.h3 />
        <form
          noValidate={true}
        >
          <Input
            EnableFixedInput={null}
            autoComplete="new-password"
            dataTest="RecoveryEmail"
            disableKeys={Array []}
            disabled={false}
            fixedValue={null}
            icon="Email"
            label={null}
            msgError={null}
            msgFocusInput={null}
            name="email"
            onBlur={[Function]}
            onChange={[MockFunction]}
            onFocus={null}
            placeholder=""
            tinyLabels={null}
            type="text"
            valid={[Function]}
            value="te000steee@asdas.com"
          />
          <styled.div>
            <Button
              actionSecondary={null}
              className={null}
              dataTest="SendButton"
              disabled={false}
              height={null}
              isCallToAction={true}
              isWarning={null}
              onClick={null}
              opacity={0.5}
              paddingSize={null}
              small={false}
              type="submit"
              width="180"
            />
          </styled.div>
          <styled.div>
            <Link
              actionSecondary={false}
              anchor={true}
              dataTest="BackToHomeScreen"
              disabled={false}
              href={null}
              isCallToAction={null}
              loading={false}
              noSpan={true}
              onClick={null}
              opacity={1}
              small={null}
              target={null}
              to="/"
              width={null}
            />
          </styled.div>
        </form>
      </styled.div>
    </styled.div>
  </styled.div>
</Fragment>
`;

exports[`RecoverPassword component should match snapshot with handleUserInput props 1`] = `
<Fragment>
  <styled.div>
    <styled.div
      data-test="RecoverPassword"
    >
      <styled.div>
        <styled.div>
          <Icon
            color={null}
            cursor={false}
            fixCrossAxisAlign={null}
            height={null}
            type="BrandLogo"
            width={null}
          />
        </styled.div>
      </styled.div>
      <styled.div>
        <styled.h2 />
        <styled.h3 />
        <form
          noValidate={true}
        >
          <Input
            EnableFixedInput={null}
            autoComplete="new-password"
            dataTest="RecoveryEmail"
            disableKeys={Array []}
            disabled={false}
            fixedValue={null}
            icon="Email"
            label={null}
            msgError={null}
            msgFocusInput={null}
            name="email"
            onBlur={[Function]}
            onChange={[MockFunction]}
            onFocus={null}
            placeholder=""
            tinyLabels={null}
            type="text"
            valid={[Function]}
            value=""
          />
          <styled.div>
            <Button
              actionSecondary={null}
              className={null}
              dataTest="SendButton"
              disabled={true}
              height={null}
              isCallToAction={true}
              isWarning={null}
              onClick={null}
              opacity={0.5}
              paddingSize={null}
              small={false}
              type="submit"
              width="180"
            />
          </styled.div>
          <styled.div>
            <Link
              actionSecondary={false}
              anchor={true}
              dataTest="BackToHomeScreen"
              disabled={false}
              href={null}
              isCallToAction={null}
              loading={false}
              noSpan={true}
              onClick={null}
              opacity={1}
              small={null}
              target={null}
              to="/"
              width={null}
            />
          </styled.div>
        </form>
      </styled.div>
    </styled.div>
  </styled.div>
</Fragment>
`;
